source/features/no-useless-split-diff-view.tsx

```diff
@@ -15,6 +15,10 @@ function isUnifiedDiff(): boolean {
 function init(): void {
 	for (const diffTable of select.all('.js-diff-table:not(.rgh-no-useless-split-diff-view-visited)')) {
 		diffTable.classList.add('rgh-no-useless-split-diff-view-visited');
+		if (!select.exists('.blob-code-addition, .blob-code-deletion', diffTable)) {
+			continue;
+		}
```

Comment:
This seems like the simplest solution to me, but I'm not sure if the check needs to happen for every diff table. Maybe once at the top of `init()` would be enough?

Reply:
Good point! However this doesn’t cover the data attribute. The selector should include that too, ideally using :is(data etc left, data etc right):is(addition, deletion) 

Since this feature does not run before dom ready, you can move this into an `exclude` so that it can be disabled completely.

Reply:
Also it needs a comment in the code to explain the selector and ensure that the condition in the code will make you change this condition too

Reply:
> you can move this into an exclude so that it can be disabled completely

This is a nitpick but isn't the usual procedure in this case to return `false` from the `init()` function? Since the feature *could* run on the page, but decides not to so it doesn't break things.

Reply:
> Since the feature _could_ run on the page, but decides not to so it doesn't break things.

That's contradictory. It's not compatible with the page, so it _cannot_ run. 

After https://github.com/sindresorhus/refined-github/pull/3105 we started preferring `exclude` regardless of the reason, in practice they're equivalent, but moving them to `exclude` is just clearer.

Ideally we would show "feature skipped because incompatible  etc etc" in the console and options page, but we don't have such a system in place. There are a couple of features that are disabled the same way in Firefox and Safari: They _could_ run, but they don't because the browser is incompatible.